src/pages/Lend/LoanChannelCategoryRecommendedByLenders.vue

```diff
@@ -0,0 +1,517 @@
+				items {
+					values {
+						id
+					}
+				}
+			}
+		}
+	}
+`;
+
+// Turn an object of graphql variables into an object of url query parameters
+function toUrlParams(variables) {
+	return _mapValues(urlParamTransform, ({ to }) => to(variables));
+}
+
+// Turn an object of url query parameters into an object of graphql variables
+function fromUrlParams(params) {
+	return _merge({}, ..._invokeMap(urlParamTransform, 'from', params));
+}
+
+function getTargetedChannel(targetedRoute, allChannels) {
+	const loanChannels = _get(allChannels, 'lend.loanChannels.values');
+	// map id from loan channels
+	const targetedLoanChannel = _filter(
+		loanChannels,
+		loanChannel => {
+			return loanChannel.url.split('/').pop() === targetedRoute;
+		}
+	);
+	// isolate targeted loan channel id
+	return _get(targetedLoanChannel[0], 'id') || null;
+}
+
+export default {
+	name: 'LoanChannelCategoryRecommendedByLenders',
+	components: {
+		LoanCardController,
+		KvPagination,
+		ViewToggle,
+	},
+	inject: ['apollo', 'cookieStore'],
+	mixins: [
+		lendFilterExpMixin,
+		loanChannelQueryMapMixin,
+	],
+	metaInfo() {
+		return {
+			link: [
+				{
+					vmid: 'canonical',
+					rel: 'canonical',
+					href: `${this.handleCanonicalUrl}`
+				}
+			]
+		};
+	},
+	data() {
+		return {
+			offset: 0,
+			limit: loansPerPage,
+			targetedLoanChannelID: null,
+			loanChannel: () => {},
+			isVisitor: true,
+			itemsInBasket: [],
+			pageQuery: { page: '1' },
+			loading: false,
+			lendFilterExpVersion: '',
+			selectedChannelLoanIds: [],
+		};
+	},
+	computed: {
+		urlParams() {
+			return toUrlParams({
+				offset: this.offset,
+			});
+		},
+		lastLoanPage() {
+			return Math.ceil(this.totalCount / this.limit);
+		},
+		loanChannelName() {
+			return _get(this.loanChannel, 'name') || 'No loans found';
+		},
+		loanChannelDescription() {
+			return _get(this.loanChannel, 'description') || null;
+		},
+		loans() {
+			const loans = this.loanChannel?.loans?.values || [];
+			const loansFiltered = loans.filter(loan => loan.repaymentInterval === 'monthly');
+			return loansFiltered;
+		},
+		loanIds() {
+			return _map(this.loans, 'id');
+		},
+		totalCount() {
+			return _get(this.loanChannel, 'loans.totalCount') || 0;
+		},
+		loanQueryVars() {
+			return {
+				ids: [this.targetedLoanChannelID],
+				limit: this.limit,
+				offset: this.offset
+			};
+		},
+		filterUrl() {
+			// initial release sends us back to /lend
+			// return `/lend/${this.$route.params.category || ''}`;
+			return this.lendFilterExpVersion === 'b'
+				? this.getAlgoliaFilterUrl()
+				: `/lend/${this.$route.params.category || ''}`;
+		},
+		handleCanonicalUrl() {
+			let url = `https://${this.$appConfig.host}${this.$route.path}`;
+			if (this.$route.query.page && Number(this.$route.query.page) > 1) {
+				url = `${url}?page=${this.$route.query.page}`;
+			}
+			return url;
+		}
+	},
+	apollo: {
+		preFetch(config, client, args) {
+			return client.query({
+				query: loanChannelPageQuery
```

Comment:
Because this component and LoanChannelCategoryControl are both imported into LoanChannelCategoryPage, both of their prefetch functions will be called. Because the prefetches are nearly identical, the prefetch could happen in LoanChannelCategoryPage instead, and the data could be passed in through props. Alternatively, LoanChannelCategoryRecommendedByLenders could be used as the route handler for /lend-by-category/recommended-by-lenders instead of falling through to using LoanChannelCategoryPage.